include/picongpu/fields/incidentField/profiles/InsightPulse.hpp

```diff
@@ -0,0 +1,618 @@
+                            int const transvAxisIdxPlusOne = static_cast<int>(pmacc::math::dot(
+                                xyzAxisIdxPlusOne,
+                                pmacc::math::cross(getDirection(), getPolarisationVector())));
+
+                            // check whether the system is right handed, i.e. direction x polarisation > 0
+                            bool rh = true; // > 0
+                            if(Unitless::propagationAxisOpenPMD == "x" and Unitless::polarisationAxisOpenPMD == "z"
+                               or Unitless::propagationAxisOpenPMD == "y" and Unitless::polarisationAxisOpenPMD == "x"
+                               or Unitless::propagationAxisOpenPMD == "z" and Unitless::polarisationAxisOpenPMD == "y")
+                                rh = false; // < 0
+
+                            // find the index in field data which is closest to totalCellIdx and timeOriginPIC
+                            float3_X idxClosestRaw; // raw = not yet rounded to integers
+
+                            for(uint32_t d = 0u; d < simDim; d++)
+                            {
+                                // return zero if transversal simulation window or simulation timestep exceeds chunk
+                                // extent
+                                if(d != propAxisIdx
+                                       and (posPIC[d] < offsetOpenPMDdataBox(d) or posPIC[d] > offsetOpenPMDdataBox(d) + (extentOpenPMDdataBox(d) - 1.0_X) * cellSizeOpenPMDdataBox(d))
+                                   or (timeOriginPIC - Unitless::TIME_DELAY) < 0.0_X
+                                   or (timeOriginPIC - Unitless::TIME_DELAY) > (extentOpenPMDdataBox(d) - 1.0_X)
+                                           * cellSizeOpenPMDdataBox(d) / SPEED_OF_LIGHT)
+                                    return 0.0_X;
+                                else
+                                {
+                                    if(d == pmacc::math::abs(polAxisIdxPlusOne) - 1 and polAxisIdxPlusOne < 0
+                                       or d == pmacc::math::abs(transvAxisIdxPlusOne) - 1
+                                           and (transvAxisIdxPlusOne < 0 and rh == true or transvAxisIdxPlusOne > 0 and rh == false))
+                                        // invert direction
+                                        idxClosestRaw[d] = extentOpenPMDdataBox(d) - 1.0_X
+                                            - (posPIC[d] - offsetOpenPMDdataBox(d)) / cellSizeOpenPMDdataBox(d);
+                                    else // keep original direction
+                                        idxClosestRaw[d]
+                                            = (posPIC[d] - offsetOpenPMDdataBox(d)) / cellSizeOpenPMDdataBox(d);
+                                }
+                            }
+
+                            // correct longitudinal index
```

Comment:
I meant it in the sense of 'fix' since the for-loop in the lines before just tackles the transversal index, but produces nonsense for the longitudinal one. This solution seemed easier to me than another case-differentiating if-statement in that (already crowded) for-loop